actionmailer/lib/action_mailer/message_delivery.rb

```diff
@@ -138,14 +138,37 @@ def enqueue_delivery(delivery_method, options = {})
         else
           job = @mailer_class.delivery_job
 
-          if job <= MailDeliveryJob
+          if use_legacy_args?(job)
             job.set(options).perform_later(
               @mailer_class.name, @action.to_s, delivery_method.to_s, args: @args)
+          elsif job <= DeliveryJob
+            job.set(options).perform_later(
+              @mailer_class.name, @action.to_s, delivery_method.to_s, *@args)
           else
+            ActiveSupport::Deprecation.warn(<<~EOM)
+              In Rails 6.2, Action Mailer will pass the mail arguments inside the `:args` keyword argument.
+              The `perform` method of the #{job} needs to change and forward the mail arguments
+              from the `args` keyword argument.
+
+              The `perform` method should now look like:
+
+              `def perform(mailer, mail_method, delivery, args:)`
+            EOM
+
             job.set(options).perform_later(
               @mailer_class.name, @action.to_s, delivery_method.to_s, *@args)
           end
         end
       end
+
+      def use_legacy_args?(job)
```

Comment:
Isn't this method name the opposite of the implementation? It returns true when the keyword argument is used, so it is not the legacy args.

Reply:
Ah yes 🤦